tests/utils.py

```diff
@@ -51,12 +59,63 @@ def __init__(self, handler: type[BaseRequestHandler]) -> None:
 @contextmanager
 def http_server(
     handler: type[BaseRequestHandler], *, tls_enabled: bool = False
-) -> Iterator[HttpServerThread]:
+) -> Iterator[HTTPServer]:
     server_cls = HttpsServerThread if tls_enabled else HttpServerThread
-    with filelock.FileLock(LOCK_PATH):
-        server = server_cls(handler)
-        server.start()
-        try:
-            yield server
-        finally:
-            server.terminate()
+    server_thread = server_cls(handler)
+    server_thread.start()
+    port = server_thread.server.server_port
+    try:
+        socket.create_connection(('localhost', port), timeout=0.5).close()
+        yield server_thread.server  # Connection has been confirmed possible; proceed.
+    finally:
+        server_thread.terminate()
+
+
+@contextmanager
+def rewrite_hyperlinks(app: Sphinx, server: HTTPServer) -> Iterator[None]:
+    """
+    Rewrite hyperlinks that refer to network location 'localhost:7777',
+    allowing that location to vary dynamically with the arbitrary test HTTP
+    server port assigned during unit testing.
+
+    :param app: The Sphinx application where link replacement is to occur.
+    :param server: Destination server to redirect the hyperlinks to.
+    """
+    match_netloc, replacement_netloc = (
+        'localhost:7777',
+        f'localhost:{server.server_port}',
+    )
+
+    def rewrite_hyperlink(_app: Sphinx, uri: str) -> str | None:
+        parsed_uri = urlparse(uri)
+        if parsed_uri.netloc != match_netloc:
+            return uri
+        return parsed_uri._replace(netloc=replacement_netloc).geturl()
+
+    listener_id = app.connect('linkcheck-process-uri', rewrite_hyperlink)
+    yield
+    app.disconnect(listener_id)
+
+
+@contextmanager
+def serve_application(
+    app: Sphinx,
+    handler: type[BaseRequestHandler],
+    *,
+    tls_enabled: bool = False,
+) -> Iterator[str]:
+    """
+    Prepare a temporary server to handle HTTP requests related to the links
+    found in a Sphinx application project.
+
+    :param app: The Sphinx application.
+    :param handler: Determines how each request will be handled.
+    :param tls_enabled: Whether TLS (SSL) should be enabled for the server.
+
```

Comment:
```suggestion
    :param tls_enabled: Whether TLS (SSL) should be enabled for the server.
	:param port: Optional server port (default: auto).
```

Reply:
Oops; made a typo when applying this.  I'll wait for CI to complete first (tests did succeed locally).